maps_dashboards/public/model/documentLayerFunctions.ts

```diff
@@ -339,6 +341,25 @@ const updateLayerConfig = (
   }
 };
 
+const clickPopup = new Popup({
+  closeButton: false,
+  closeOnClick: false,
+});
+export const getClickPopup = () => {
+  return clickPopup;
+};
+
+export const buildPopup = (features: any[], title: string) => {
+  const div = document.createElement('div');
+  ReactDOM.render(
```

Comment:
Nit, ReactDOM.render() is no longer supported in React 18 although OSD currently using React 16.

We might can replace it, for example:
```
import { useEffect, useRef } from 'react';


export const MyComponent = (props) => {
  const clickPopup = new Popup({
    closeButton: false,
    closeOnClick: false,
  });

  const popupRef = useRef();

  const updatePopup = (features, title) => {
    // create a new div element
    const div = document.createElement('div');
    // render the TooltipClick component into the div element
    React.createElement(
      TooltipClick,
      { title, features, onClick: () => clickPopup.remove() },
      null
    );
    // update the ref with the new div element
    popupRef.current = div;
    // update the Popup object's DOM content and coordinates
    clickPopup.setDOMContent(div).setLngLat(features[0].geometry.coordinates);
  };

  useEffect(() => {
    updatePopup(props.features, props.title);
  }, [props.features, props.title]); // this will cause the effect to run whenever props.features or props.title changes

  return (
    // other JSX elements
  );
};

```

Reply:
Will work on this later.